### PR TITLE
chore(io): Introduce (carefully) new io with use_new_io flag

### PR DESCRIPTION
**Pull request by dranikpg**

### CODE REVIEW

File: src/facade/reply_builder.cc [hunk 1]
```diff
@@ -546,10 +542,10 @@ void RedisReplyBuilder::SendScoredArray(const std::vector<std::pair<std::string,
       if (indx % 2 == 0)
         return arr[indx / 2].first;
 
-      // NOTE: we reuse the same buffer, assuming that SendStringArrInternal does not reference
-      // previous string_views. The assumption holds for small strings like
-      // doubles because SendStringArrInternal employs small string optimization.
-      // It's a bit hacky but saves allocations.
+      // NOTE: we reuse the same buffer, assuming that SendStringArrInternal
```
Comment (romange) [inline src/facade/reply_builder.cc:545]: what's up with all the spaces changes in this file?

File: src/facade/reply_builder.cc [hunk 2]
```diff
@@ -608,7 +605,8 @@ void RedisReplyBuilder::SendMGetResponse(MGetResponse resp) {
   auto get_pending_metabuf = [&] { return string_view{cur_meta, size_t(next - cur_meta)}; };
 
   for (unsigned i = 0; i < size; ++i) {
-    DCHECK_GE(meta.end() - next, 16);  // We have at least 16 bytes for the meta data.
+    DCHECK_GE(meta.end() - next,
```
Comment (romange) [inline src/facade/reply_builder.cc:608]: why this change was made, for example?

File: tests/analyze.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
+from collections import defaultdict
```
Comment (romange) [inline tests/analyze.py:1]: what's this?

File: tests/dragonfly/instance.py [hunk 4]
```diff
@@ -87,9 +87,6 @@ def __init__(self, params: DflyParams, args):
             if threads > 1:
                 self.args["num_shards"] = threads - 1
 
-        # Add 1 byte limit for big values
```
Comment (romange) [inline tests/dragonfly/instance.py:90]: seems like a bad merge?

File: tests/out.txt [hunk 5]
```diff
@@ -0,0 +1,1646 @@
+============================= test session starts ==============================
```
Comment (romange) [inline tests/out.txt:1]: not needed

File: tests/analyze.py (hunk 3 above)
Comment (dranikpg) [reply tests/analyze.py:1]: Yes, sorry, I was blindly pushing for CI, forgot to untag your review

File: src/facade/reply_builder_test.cc [hunk 6]
```diff
@@ -205,8 +205,8 @@ RedisReplyBuilderTest::ParsingResults RedisReplyBuilderTest::Parse() {
 
 TEST_F(RedisReplyBuilderTest, MessageSend) {
   // Test each message that is "sent" to the sink
-  builder_->SendOk();
-  ASSERT_EQ(TakePayload(), kOKMessage);
+  // builder_->SendOk();
```
Comment (romange) [inline src/facade/reply_builder_test.cc:208]: todo: to bring it back or restore?

**Review by romange — APPROVED** [review]

File: src/facade/reply_builder_test.cc (hunk 6 above)
Comment (dranikpg) [reply src/facade/reply_builder_test.cc:208]: Yes, actually I'll be simpler to just revert this commit

Comment (romange) [reply src/facade/reply_builder_test.cc:208]: but you have not,  @dranikpg ?

**Review by romange — COMMENTED** [review]
lgtm but still not sure what happens in MessageSend test

**Review by romange — APPROVED** [review]
